python/vyos/ifconfig/wireguard.py

```diff
@@ -156,6 +157,63 @@ def show_interface(self):
                 answer += '\n'
         return answer
 
+    def get_latest_handshakes(self):
+        """Get latest handshake time for each peer"""
+        output = {}
+
+        # Dump wireguard last handshake
+        _f = self._cmd(f'wg show {self.ifname} latest-handshakes')
+        # Output:
+        # xxxw=    1732812147
+        # xxx=    0
+        for line in _f.split('\n'):
+            if not line:
+                # Skip empty lines and last line
+                continue
+            items = line.split('\t')
+
+            if len(items) != 2:
+                continue
+
+            output[items[0]] = int(items[1])
+
+        return output
+
+    def reset_peer(self, peer_name=None, public_key=None):
+        from vyos.configquery import ConfigTreeQuery
+        c = ConfigTreeQuery()
+
+        max_dns_retry = c.value(['interfaces', 'wireguard', self.ifname, 'max-dns-retry'])
+        if max_dns_retry is None:
+            max_dns_retry = 3
+
+        for peer in c.list_nodes(['interfaces', 'wireguard', self.ifname, 'peer']):
+            peer_public_key = c.value(['interfaces', 'wireguard', self.ifname, 'peer', peer, 'public-key'])
+            if peer_name is None or peer == peer_name or public_key == peer_public_key:
+                address = c.value(['interfaces', 'wireguard', self.ifname, 'peer', peer, 'address'])
+                port = c.value(['interfaces', 'wireguard', self.ifname, 'peer', peer, 'port'])
+
+                if not address or not port:
+                    if peer_name is not None:
+                        print(f'Peer {peer_name} endpoint not set')
+                    continue
+
+                if c.exists(['interfaces', 'wireguard', self.ifname, 'peer', peer, 'disable']):
+                    continue
+
+                cmd = f"wg set {self.ifname} peer {peer_public_key} endpoint {address}:{port}"
+                try:
+                    print(
+                        f'Resetting {self.ifname} peer {peer_public_key} endpoint to {address}:{port} ... ',
```

Comment:
Can you please output the previous endpoint IP address and the new endpoint IP address as well?